### PR TITLE
Added functions that allow you to look at the emits parameters

### PR DESCRIPTION
**Pull request by brrmorre**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: API modification

**Development Phase**: implementation

**Release Notes**

Added functions that allow you to look at the emits parameters. I added these so that the monitor could look at both the emits and supports parameters.

### CODE REVIEW

File: src/main/scala/tilelink/Parameters.scala [hunk 1]
```diff
@@ -1390,6 +1439,15 @@ case class TLEdgeParameters(
   // Sanity check the link...
   require (maxTransfer >= slave.beatBytes, s"Link's max transfer (${maxTransfer}) < ${slave.slaves.map(_.name)}'s beatBytes (${slave.beatBytes})")
 
+  def anyDiplomaticClaimsMasterToSlaveAcquireT   = master.anyEmitClaimsAcquireT   || slave.anySupportAcquireT
```
Comment (jchang0) [inline src/main/scala/tilelink/Parameters.scala:1442]: I don't think this def is useful.  What would be useful is the && version (so erase the "any" part of this def and use && instead)

File: src/main/scala/tilelink/Parameters.scala [hunk 2]
```diff
@@ -557,8 +557,30 @@ class TLSlavePortParameters private(
...
+  // Diplomatically determined operation sizes emitted by all outward Slaves
+  // as opposed to expectsVipChecker which generate circuitry to check which specific addresses
+  val allEmitClaims = slaves.map(_.emits).reduce( _ intersect _)
+  val allEmitClaimsAcquireT   = allEmitClaims.probe
+  val allEmitClaimsArithmetic = allEmitClaims.arithmetic
+  val allEmitClaimsLogical    = allEmitClaims.logical
+  val allEmitClaimsGet        = allEmitClaims.get
+  val allEmitClaimsPutFull    = allEmitClaims.putFull
+  val allEmitClaimsPutPartial = allEmitClaims.putPartial
+  val allEmitClaimsHint       = allEmitClaims.hint
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala:569]: Please don't add these helper `val`s.

Comment (terpstra) [reply src/main/scala/tilelink/Parameters.scala:569]: ie: remove 563-569.

File: src/main/scala/tilelink/Parameters.scala [hunk 3]
```diff
@@ -557,8 +557,30 @@ class TLSlavePortParameters private(
...
+  // Operation Emitted by at least one outward Slaves
+  // as opposed to expectsVipChecker which generate circuitry to check which specific addresses
+  val anyEmitClaims = slaves.map(_.emits).reduce(_ cover _)
+  val anyEmitClaimsAcquireT   = !anyEmitClaims.probe.none
+  val anyEmitClaimsArithmetic = !anyEmitClaims.arithmetic.none
+  val anyEmitClaimsLogical    = !anyEmitClaims.logical.none
+  val anyEmitClaimsGet        = !anyEmitClaims.get.none
+  val anyEmitClaimsPutFull    = !anyEmitClaims.putFull.none
+  val anyEmitClaimsPutPartial = !anyEmitClaims.putPartial.none
+  val anyEmitClaimsHint       = !anyEmitClaims.hint.none
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala:580]: Remove 574-580.

File: src/main/scala/tilelink/Parameters.scala [hunk 4]
```diff
@@ -1119,7 +1142,32 @@ class TLMasterPortParameters private(
...
+  // as opposed to expectsVipChecker which generate circuitry to check which specific addresses
+  val allEmitClaims = masters.map(_.emits).reduce( _ intersect _)
+  val allEmitClaimsAcquireT   = allEmitClaims.acquireT
+  val allEmitClaimsAcquireB   = allEmitClaims.acquireB
+  val allEmitClaimsArithmetic = allEmitClaims.arithmetic
+  val allEmitClaimsLogical    = allEmitClaims.logical
+  val allEmitClaimsGet        = allEmitClaims.get
+  val allEmitClaimsPutFull    = allEmitClaims.putFull
+  val allEmitClaimsPutPartial = allEmitClaims.putPartial
+  val allEmitClaimsHint       = allEmitClaims.hint
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala:1155]: Remove 1148-1155.

File: src/main/scala/tilelink/Parameters.scala [hunk 5]
```diff
@@ -1119,7 +1142,32 @@ class TLMasterPortParameters private(
...
+  // as opposed to expectsVipChecker which generate circuitry to check which specific addresses
+  val anyEmitClaims = masters.map(_.emits).reduce(_ cover _)
+  val anyEmitClaimsAcquireT   = !anyEmitClaims.acquireT.none
+  val anyEmitClaimsAcquireB   = !anyEmitClaims.acquireB.none
+  val anyEmitClaimsArithmetic = !anyEmitClaims.arithmetic.none
+  val anyEmitClaimsLogical    = !anyEmitClaims.logical.none
+  val anyEmitClaimsGet        = !anyEmitClaims.get.none
+  val anyEmitClaimsPutFull    = !anyEmitClaims.putFull.none
+  val anyEmitClaimsPutPartial = !anyEmitClaims.putPartial.none
+  val anyEmitClaimsHint       = !anyEmitClaims.hint.none
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala:1167]: Remove 1160-1167.

File: src/main/scala/tilelink/Parameters.scala [hunk 6]
```diff
@@ -1390,6 +1439,15 @@ case class TLEdgeParameters(
...
 
+  def diplomaticClaimsMasterToSlaveAcquireT   = master.anyEmitClaimsAcquireT   && slave.anySupportAcquireT
+  def diplomaticClaimsMasterToSlaveAcquireB   = master.anyEmitClaimsAcquireB   && slave.anySupportAcquireB
+  def diplomaticClaimsMasterToSlaveArithmetic = master.anyEmitClaimsArithmetic && slave.anySupportArithmetic
+  def diplomaticClaimsMasterToSlaveLogical    = master.anyEmitClaimsLogical    && slave.anySupportLogical
+  def diplomaticClaimsMasterToSlaveGet        = master.anyEmitClaimsGet        && slave.anySupportGet
+  def diplomaticClaimsMasterToSlavePutFull    = master.anyEmitClaimsPutFull    && slave.anySupportPutFull
+  def diplomaticClaimsMasterToSlavePutPartial = master.anyEmitClaimsPutPartial && slave.anySupportPutPartial
+  def diplomaticClaimsMasterToSlaveHint       = master.anyEmitClaimsHint       && slave.anySupportHint
+
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala:1450]: Maybe master.anyEmit.intersect(slave.anySupport).

**Pull request merged**